arangod/Utils/CollectionNameResolver.cpp

```diff
@@ -32,7 +32,19 @@
 #include "VocBase/LogicalView.h"
 #include "VocBase/vocbase.h"
 
-using namespace arangodb;
+namespace arangodb {
+
+std::shared_ptr<LogicalCollection> CollectionNameResolver::getCollection(
+  TRI_voc_cid_t id
+) const noexcept {
+  return std::dynamic_pointer_cast<LogicalCollection>(getDataSource(id));
+}
+
+std::shared_ptr<LogicalCollection> CollectionNameResolver::getCollection(
+  std::string const& nameOrId
+) const noexcept {
+  return std::dynamic_pointer_cast<LogicalCollection>(getDataSource(nameOrId));
```

Comment:
why `dynamic_pointer_cast`?

Reply:
because getDataSource(id) returns a std::shared_pointer which can actually be a LogicalCollection or a LogicalView

Reply:
please use something like:
```
auto dataSource = getDataSource(nameOrId);
return dataSource.category() == LogicalCollection::category() ? std::static_pointer_cast<LogicalCollection>(dataSource) : nullptr;
```
instead

Reply:
done